### PR TITLE
doc: add other file types used in detection

### PR DESCRIPTION
**Pull request by mhdawson**
We added support for .cjs and .mjs extensions but
missed adding to the doc.

## Summary

Add missing documentation.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by c0d1ngm0nk3y** [comment]
I don't think we should merge this yet, since this requires `libnodejs` `v0.3.0` and `node-start` still uses `v0.2.0`. Not sure why this was not auto bumped though.

File: README.md [hunk 1]
```diff
@@ -48,11 +48,19 @@ This could be useful if your app is a part of a monorepo.
 
 This buildpack searches your application root for the following files:
 1. `server.js`
+1. `server.cjs`
+1. `server.mjs`
```
Comment (c0d1ngm0nk3y) [inline README.md:52]:
```suggestion
1. `server.[c|m]js`
```

File: README.md [hunk 2]
```diff
@@ -48,11 +48,19 @@ This could be useful if your app is a part of a monorepo.
...
 1. `app.js`
+1. `app.cjs`
+1. `app.mjs`
```
Comment (c0d1ngm0nk3y) [inline README.md:55]:
```suggestion
1. `app.[c|m]js`
```

File: README.md [hunk 3]
```diff
@@ -48,11 +48,19 @@ This could be useful if your app is a part of a monorepo.
...
 1. `main.js`
+1. `main.cjs`
+1. `main.mjs`
```
Comment (c0d1ngm0nk3y) [inline README.md:58]:
```suggestion
1. `main.[c|m]js`
```

File: README.md [hunk 4]
```diff
@@ -48,11 +48,19 @@ This could be useful if your app is a part of a monorepo.
...
 1. `index.js`
+1. `index.cjs`
+1. `index.mjs`
```
Comment (c0d1ngm0nk3y) [inline README.md:61]:
```suggestion
1. `index.[c|m]js`
```

**Review by c0d1ngm0nk3y — COMMENTED** [review]
Maybe have a look at my suggestions. It is less exact (no order within a basename), but maybe easier to read? WDYT?

**Comment by mhdawson** [comment]
@c0d1ngm0nk3y accepted your suggestions.

**Comment by mhdawson** [comment]
@pacostas if you could take a look and approve/land this one it would be great.

**Review by c0d1ngm0nk3y — APPROVED** [review]

**Pull request merged**